staging/src/k8s.io/component-helpers/apps/poddisruptionbudget/helpers.go

```diff
@@ -28,21 +28,21 @@ func UpdateDisruptionAllowedCondition(pdb *policy.PodDisruptionBudget) {
 	if pdb.Status.Conditions == nil {
 		pdb.Status.Conditions = make([]metav1.Condition, 0)
 	}
+
+	condition := metav1.Condition{
+		Type:               policy.DisruptionAllowedCondition,
+		ObservedGeneration: pdb.Status.ObservedGeneration,
+	}
+
 	if pdb.Status.DisruptionsAllowed > 0 {
-		apimeta.SetStatusCondition(&pdb.Status.Conditions, metav1.Condition{
-			Type:               policy.DisruptionAllowedCondition,
-			Reason:             policy.SufficientPodsReason,
-			Status:             metav1.ConditionTrue,
-			ObservedGeneration: pdb.Status.ObservedGeneration,
-		})
+		condition.Reason = policy.SufficientPodsReason
+		condition.Status = metav1.ConditionTrue
 	} else {
-		apimeta.SetStatusCondition(&pdb.Status.Conditions, metav1.Condition{
-			Type:               policy.DisruptionAllowedCondition,
-			Reason:             policy.InsufficientPodsReason,
-			Status:             metav1.ConditionFalse,
-			ObservedGeneration: pdb.Status.ObservedGeneration,
-		})
+		condition.Reason = policy.InsufficientPodsReason
+		condition.Status = metav1.ConditionFalse
 	}
+
+	apimeta.SetStatusCondition(&pdb.Status.Conditions, condition)
```

Comment:
It's arguable if the refactoring makes the code more readable. IMO the scattered condition config makes it less clear what is set. Please revert.

Reply:
> It's arguable if the refactoring makes the code more readable. IMO the scattered condition config makes it less clear what is set. Please revert.
already rolled back.